### PR TITLE
Switch base docker image from golang-alpine to ubuntu:20.04

### PR DESCRIPTION
**Pull request by jkneubuh**
#### Type of change

- Bug fix

#### Description

This PR switches the Fabric base docker images from golang-alpine to ubuntu:20.04.  The upgrade is necessary as the golang-alpine libc runtimes (musl) are incompatible with the requirements on multi-arch runtimes. 

The motivations for this upgrade are described in detail in Discussion #3876 and comments in DRAFT PR #3877 

#### Additional details

- Discussion #3876 
- PR #3877  (DRAFT)

#### Related issues

- #2994

### CODE REVIEW

**Comment by jkneubuh** [comment]
@denyeart @C0rWin @pfi79 @davidkel @lehors - final stake / compromise for arm64 support.

File: images/ccenv/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
-RUN mkdir -p /chaincode/output /chaincode/input
-RUN addgroup -g 500 chaincode && adduser -u 500 -D -h /home/chaincode -G chaincode chaincode
-RUN chown -R chaincode:chaincode /chaincode
+
+RUN apt update
+RUN apt install -y binutils-gold
+RUN apt install -y curl
+RUN apt install -y g++
+RUN apt install -y gcc
+RUN apt install -y git
```
Comment (C0rWin) [inline images/ccenv/Dockerfile]: Why not to group with single RUN?

File: images/tools/Dockerfile [hunk 2]
```diff
@@ -1,41 +1,76 @@
...
-VOLUME /etc/hyperledger/fabric
-COPY --from=tools /go/src/github.com/hyperledger/fabric/build/bin /usr/local/bin
-COPY --from=tools /go/src/github.com/hyperledger/fabric/sampleconfig ${FABRIC_CFG_PATH}
+
+RUN apt update
+RUN apt install -y bash
+RUN apt install -y curl
+RUN apt install -y jq
+RUN apt install -y tzdata
+
```
Comment (C0rWin) [inline images/tools/Dockerfile]: this also could be grouped together

File: images/tools/Dockerfile [hunk 3]
```diff
@@ -1,41 +1,76 @@
...
+FROM ubuntu:${UBUNTU_VER}
+
+ARG TARGETOS
+ARG TARGETARCH
 ARG FABRIC_VER
-ENV FABRIC_VER ${FABRIC_VER}
-
-# Disable cgo when building in the container.  This will create a static binary, which can be
-# copied and run in the base alpine container without the libc/musl runtime.
-ENV CGO_ENABLED 0
```
Comment (C0rWin) [inline images/tools/Dockerfile:27]: Why did you remove static compilation?

Comment (jkneubuh) [reply images/tools/Dockerfile:27]: The static link in docker was only added to work around the SIGSEGVs in alpine.  Now with the build in Docker and with the matching libc runtime FROM ubuntu, the static build is no longer required.

**Comment by jkneubuh** [comment]
Looks like ubuntu:20.04 has been on a tighter diet than debian:slim.  This is on the arm64 - also on amd64 the debian images are a few MB larger. 

```
jkneubuh@gala fabric % docker images 
debian    stable-slim    86f9b934c377   22 hours ago    74.3MB
ubuntu    20.04          43ed104e759f   12 days ago     65.7MB
alpine    latest         d3156fec8bcb   4 weeks ago     7.46MB
```

File: images/tools/Dockerfile (hunk 3 above)
Comment (C0rWin) [reply images/tools/Dockerfile:27]: Compiling binary statically has another side effect of having all dependencies compiled into binary, ensuring the resulting executable is assembled and equipped with all dependencies it was compiled with, ensuring consistency.

Comment (jkneubuh) [reply images/tools/Dockerfile:27]: @C0rWin are you proposing that we change the `make` target to generate a static executable?  I thought we had settled on the use of Docker to establish consistency across runtimes.  Recall that the above update `ENV CGO_ENABLED 0` was _new behavior_ introduced in the release 2.5 alpha builds for alpine - the behavior in this PR reverts to the same mechanisms we've been using in Fabric since ~2017, with the primary difference that the base Docker image is now derived from ubuntu.  Running with the ubuntu / gnu libc seems to have resolved all of the errors encountered with alpine's musl runtime.

Also: 

- Formal release binaries are static, generated with the golang cross-compiler.  

- Changing the default `make` target to a static build will cause major issues for the pkcs11 runtimes.

**Review by denyeart — APPROVED** [review]
Looks like there is general agreement in the Discussion on the approach. Let's give it a try!

**Comment by denyeart** [comment]
@Mergifyio backport main
